lib/segment/src/segment.rs

```diff
@@ -181,15 +194,32 @@ impl Segment {
         check_named_vectors(&vectors, &self.segment_config)?;
         for (vector_name, new_vector) in vectors {
             let vector_data = &self.vector_data[vector_name.as_ref()];
+            let old_vector = if new_vector.as_vec_ref().is_sparse() {
+                // Sparse vector index requires an old vector to be passed
+                // to update operation, so it can clean up old index entries properly
+                //
+                // So we retrieve it here.
+                vector_data
+                    .vector_storage
+                    .borrow()
```

Comment:
Performance nit: maybe borrow `vector_storage` once per loop instead of borrowing it twice for each loop iteration?